src/simple_http.rs

```diff
@@ -19,13 +20,55 @@ use serde_json;
 use crate::client::Transport;
 use crate::{Request, Response};
 
+#[cfg(fuzzing)]
+/// Global mutex used by the fuzzing harness to inject data into the read
+/// end of the TCP stream.
+pub static FUZZ_TCP_SOCK: Mutex<Option<io::Cursor<Vec<u8>>>> = Mutex::new(None);
+
+#[cfg(fuzzing)]
+#[derive(Clone, Debug)]
+struct TcpStream;
+
+#[cfg(fuzzing)]
+mod impls {
```

Comment:
What is the reason to put the implementation of the fuzzed TcpStream into its own module?

I'm new to Rust so forgive me if this is a silly question.

Reply:
So that the `#[cfg(fuzzing)]` is applied to the whole module.

Reply:
thanks, that makes sense. 

The `TcpStream` is  declared outside of the `impls` mod so it doesn't have have to be imported explicitly in the `simple_http` mod.